src/Avalonia.Controls/ListBox.cs

```diff
@@ -128,17 +129,18 @@ protected override IItemContainerGenerator CreateItemContainerGenerator()
         }
 
         /// <inheritdoc/>
-        protected override void OnGotFocus(GotFocusEventArgs e)
+        protected override void OnGotFocus(RoutedEventArgs e)
         {
             base.OnGotFocus(e);
 
-            if (e.NavigationMethod == NavigationMethod.Directional)
+            // TODO_FOCUS:
+            if (FocusState == FocusState.Keyboard)
             {
-                e.Handled = UpdateSelectionFromEventSource(
```

Comment:
From what I understand this is doing (and same occurs in TreeView) when ListBox gets focus it sets the selected item. I don't think that's how this should work. Selection should be done by explicit interaction (either keyboard or pointer). UWP does have the `SingleSelectionFollowsFocus` property on some controls to keep focus/selection in sync for single selection mode - but that works opposite this (selection then focus). I know I saw a unit test about this too - I think this is a niche behavior rather than what would be generally expected as this forces selection to follow focus rather than letting it be an optional behavior.

Reply:
Yeah that's definitely not how it should work! I keep meaning to refactor this but never get time to finish it off.